### DISCUSSION TITLE
New baselines for dual and config

### DISCUSSION THREAD

**Discussion by gasse** [ds4dm/ml4co-competition#41] [Announcements]
Dear Participants,

Thanks to the contributions @lascavana and @antoniach, we now have two baseline methods for the dual and the config tasks! :tada:

**Dual task - imitation of the Strong Branching expert**

This baseline aims at imitating the branching decisions of the Strong Branching (SB) expert rule (see [Achterberg, sec. 5.4](https://doi.org/10.14279/depositonce-1634)), via machine learning, with a Graph Neural Network (GNN) model (see [Gasse et al.](https://papers.nips.cc/paper/2019/hash/d14c2267d848abeb81fd590f371d39bd-Abstract.html)). The proposed approach consists of two steps:
1. collect a training dataset, that is, SB scores and branching decisions
2. train the GNN with supervised learning, that is, minimize the cross-entropy loss by gradient descent

Code and readme file [here](https://github.com/ds4dm/ml4co-competition/tree/main/baseline/dual).

**Configuration task - parameter tuning with SMAC**

This baseline aims at finding a fixed set of good parameter values for each problem benchmark, by using the hyperparameter tuning tool [SMAC](https://www.automl.org/automated-algorithm-design/algorithm-configuration/smac/). In particular, the baseline uses SMAC4HPO, which performs Bayesian optimization using a Random Forest model. The extensive set of tuned [SCIP parameters](https://www.scipopt.org/doc/html/PARAMETERS.php) is the following:
```
NAME                        TYPE         RANGE           DEFAULT
branching/scorefunc         categorical  {s,p,q}         [p]
branching/scorefac          real         [0,1]           [0.167]
branching/preferbinary      categorical  {TRUE,FALSE}    [FALSE]
branching/clamp             real         [0,0.5]         [0.2]
branching/midpull           real         [0,1]           [0.75]
branching/midpullreldomtrig real         [0,1]           [0.5]
branching/lpgainnormalize   categorical  {d,l,s}         [s]
lp/pricing                  categorical  {l,a,f,p,s,q,d} [l]
lp/colagelimit              integer      [-1,2147483647] [10]
lp/rowagelimit              integer      [-1,2147483647] [10]
nodeselection/childsel      categorical  {d,u,p,i,l,r,h} [h]
separating/minortho         real         [0,1]           [0.9]
separating/minorthoroot     real         [0,1]           [0.9]
separating/maxcuts          integer      [0,2147483647]  [100]
separating/maxcutsroot      integer      [0,2147483647]  [2000]
separating/cutagelimit      integer      [-1,2147483647] [80]
separating/poolfreq         integer      [-1,65534]      [10]
```

Code and readme file [here](https://github.com/ds4dm/ml4co-competition/tree/main/baseline/config).

Please feel free to have a look at these codes, reuse them and extend them as you wish.

Good luck with your submissions,

Best,
The organizing team
